src/UCTNode.cpp

```diff
@@ -271,7 +271,7 @@ UCTNode* UCTNode::uct_select_child(int color, bool is_root) {
     auto best_value = std::numeric_limits<double>::lowest();
 
     for (auto& child : m_children) {
-        if (!child.active()) {
+        if (!child.active() || (child.is_inflated() && child->m_is_expanding)) {
```

Comment:
You will now crash if there is only one valid successor. (or assert at best != nullptr).

I'd set winrate to 0.0 or something, like a huge virtual loss.

Reply:
Does `- fpu_reduction - 1.0` make sense? I feel 0.0 is not enough since the eval may give 0 win rate in extreme conditions and we're using `parent - fpu_reduction` for unvisited nodes.

Reply:
As long as it's a very low value it will work. But I'm not sure we care about MCTS scaling if the critical path has all 0.0 winrates.

Reply:
We usually don't. But current batching implementation requires that the expanding node not being selected again, or the full batch waiting will be stuck.